src/payment/strategies/sdk/strategy.ts

```diff
@@ -0,0 +1,155 @@
+
+            if methodId is 'card', how do we differentiate between standard and "has own JS" implementations?
+
+            N.B. looking for a "has own JS" initializer, not finding one,
+            then falling back to a default one for that methodId
+            could produce some unpredictable (and likely broken) UX during execute
+        */
+        const initializer = initializers[methodId];
+
+        if (!initializer) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        // Suggestions for better names welcome :D
+        const paymentProcessor = await initializer({
+            options,
+            store: this._store,
+        });
+
+        if (!paymentProcessor) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        // Sore the processor for later use in execute
+        this._paymentProcessor = paymentProcessor;
+
+        return this._store.getState();
+    }
+
+    deinitialize(_options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        // Nothing to be done here?
+        return Promise.resolve(this._store.getState());
+    }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment || !payment.methodId) {
+            throw new PaymentArgumentInvalidError(['payment.methodId']);
+        }
+
+        const paymentProcessor = this._paymentProcessor;
+
+        if (!paymentProcessor) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        // Submitting the order should let us restore it post redirect in finalize
+        const submitOrder = () => this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+        const getPaymentStep = () => paymentProcessor(payment);
+        const loadOrder = () => this._store.dispatch(this._orderActionCreator.loadCurrentOrder());
+
+        // TODO: move to async/await format?
+        return submitOrder()
+            // Very key differences to a typical strategy:
+            .then(getPaymentStep)
+            .then(handler)
+            .then(loadOrder);
+    }
+
+    finalize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        /*
+            This is actually called first, before initialize
+            We can check if there was a payment left for us to continue
+            i.e. at the return of a redirect
+
+            N.B. we will need a different mechanism for continue steps in the future
+            'finalize' runs and dead-ends prior to rendering payment methods
+            some future steps will want to run with the method choice visually open
+            we'll likely need a new 'continueOrderIfNeeded' feature in checkout-js
+        */
```

Comment:
Considering how these new methods operate I don't think we could use `finalize` in the way it was intended previously, we'll just have to instantiate the same method and keep running `continue`s... 🤷

We might be able to change finalize to allow it to keep running and be just some sort of _pick it up where it was left_ for SDK providers... 😕

Reply:
Yeah, this is bending `finalize` purpose quite a bit. The plan is to use it only for phase 2.1, then to make an alternative lifecycle method within `checkout-js` that is better suited to 'continuing' and use that instead.